src/app/my-info/(account)/coupons/page.tsx

```diff
@@ -0,0 +1,33 @@
+import classNames from 'classnames/bind';
+import { HydrationBoundary, QueryClient, dehydrate } from '@tanstack/react-query';
+import { getCoupons } from '@/api/coupon';
+import { MyInfoEmptyCase } from '@/app/my-info/_components';
+import CouponList from './_components/CouponList';
+
+import styles from './page.module.scss';
+
+const cn = classNames.bind(styles);
+
+export default async function CouponsPage() {
+  const queryClient = new QueryClient();
+
+  await queryClient.prefetchQuery({
+    queryKey: ['coupons'],
+    queryFn: getCoupons,
+  });
+
+  const coupons = queryClient.getQueryData(['coupons']);
```

Comment:
https://hjk329.github.io/fetchQuery-vs-prefetchQuery/
https://velog.io/@baby_dev/React-Query-fetchQuery-vs-prefetchQuery
특별한 이유가 없다면 prefetch가 더 나은걸로 알고있습니다!

Reply:
fetchQuery is an asynchronous method that can be used to fetch and cache a query. It will either resolve with the data or throw with the error. Use the prefetchQuery method if you just want to fetch a query without needing the result.

결과 값을 가져오지 않을 때는 prefetchQuery를 사용하지만, 결과 값을 사용할 때는 fetchQuery를 사용하는 것 같아요~
공식 문서에서는 단순히 데이터를 caching만 해오고, 사용하는 부분이 없어서 예시에는 없는 것으로 나옵니다.(더 나은 지는 모르겠네요. 오히려 코드양을 줄인다는점[자바스크립트 번들 사이즈를 줄이는 점?]에서 이게 더 낫지 않을까요...?)

Reply:
아마 캐싱방법도 다를거에요 fetchQuery는 캐싱이 아닌 query가 마운트 시 무조건 실행되는걸로 알고있어요

Reply:
prefetchQuery 뜯어보면 코드가 그냥 아래처럼 되어있는데, 같은 거 아닌가요?
```typescript
prefetchQuery<
    TQueryFnData = unknown,
    TError = DefaultError,
    TData = TQueryFnData,
    TQueryKey extends QueryKey = QueryKey,
  >(
    options: FetchQueryOptions<TQueryFnData, TError, TData, TQueryKey>,
  ): Promise<void> {
    return this.fetchQuery(options).then(noop).catch(noop)
  }
```

Reply:
내부로직까지는 정확히 모르겠지만 확실하게 알고있는 내용으로 말씀드리면
prefetch는 성공만 쿼리에 대해서만 캐싱만 진행하는거라 return값이 없어요 또 아직 살아있는 쿼리키를 건드리게되면 fetching이 진행되지 않죠
하지만 fetchQuery는 컴포넌트 마운트시 무조건 쿼리를 실행하고 return값이 존재하는걸로 알고있어요 그래서 에러를 받아서 넘겨주거나 return값을 받아서 할수있는 동작이 가능한걸로 알고있습니다.